src/Extension/KernelExtensions.cs

```diff
@@ -13,29 +15,45 @@ namespace Extension
 {
     public static class KernelExtensions
     {
-        public static T UseQuestionMagicCommand<T>(this T kernel, Evaluator evaluator) where T : Kernel
+        public static Task<KernelCommandResult[]> SubmitEvaluationCriteriaAsync<T>(
```

Comment:
Async methods should always support cancellation by passing through a `CancellationToken`.

Reply:
Done